src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/tests/test_batch_job_commands.py

```diff
@@ -0,0 +1,109 @@
+    BatchDataPlaneTestBase)
+
+
+class BatchJobTest(BatchDataPlaneTestBase):
+
+    def __init__(self, test_method):
+        super(BatchJobTest, self).__init__(__file__, test_method)
+        self.pool_paas = "azure-cli-test-paas"
+        self.job1 = "cli-test-job-1"
+        self.data_dir = os.path.join(
+            os.path.dirname(__file__), 'data', 'batch-job-{}.json').replace('\\', '\\\\')
+
+    def tear_down(self):
+        # Clean up any running pools in case the test exited early
+        try:
+            self.cmd('batch pool delete --pool-id {} --yes'.format(self.pool_paas))
+        except Exception:  # pylint: disable=broad-except
+            pass
+        try:
+            self.cmd('batch job delete --job-id {} --yes'.format(self.job1))
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    def test_batch_jobs(self):
+        self.execute()
+
+    def body(self):
+        # pylint: disable=too-many-statements
+        # test create paas pool using parameters
+        self.cmd('batch pool create --id {} --vm-size small --os-family 4'.format(
+            self.pool_paas))
+
+        start_time = datetime.datetime.now().isoformat()
+        # test create job with missing parameters
+        try:
+            self.cmd('batch job create --id {} --metadata test=value '
+                     '--job-max-task-retry-count 5 '
+                     '--job-manager-task-run-elevated'.format(self.job1))
+        except SystemExit as exp:
+            self.assertEqual(exp.code, 2)
+
+        # test create job
+        self.cmd('batch job create --id {} --metadata test=value --job-max-task-retry-count 5 '
+                 '--job-manager-task-run-elevated --job-manager-task-id JobManager '
+                 '--job-manager-task-command-line "cmd /c set AZ_BATCH_TASK_ID" '
+                 '--job-manager-task-environment-settings '
+                 'CLI_TEST_VAR=CLI_TEST_VAR_VALUE --pool-id {}'.format(self.job1, self.pool_paas))
+
+        # test get job
+        job1 = self.cmd('batch job show --job-id {}'.format(self.job1))
+        self.assertEqual(job1['metadata'][0]['name'], 'test')
+        self.assertEqual(job1['metadata'][0]['value'], 'value')
+        self.assertEqual(job1['jobManagerTask']['environmentSettings'][0]['name'],
+                         'CLI_TEST_VAR')
+        self.assertEqual(job1['jobManagerTask']['environmentSettings'][0]['value'],
+                         'CLI_TEST_VAR_VALUE')
+        self.assertEqual(job1['jobManagerTask']['id'], 'JobManager')
+        self.assertEqual(job1['constraints']['maxTaskRetryCount'], 5)
+        self.assertEqual(job1['onAllTasksComplete'], 'noAction')
+
+        # test bad enum value
+        try:
+            self.cmd('batch job set --job-id {} '
+                     '--on-all-tasks-complete badValue '.format(self.job1))
+            raise AssertionError('Expected SystemExit to be raised.')
```

Comment:
`with self.assertRaises(SystemExit)` should be sufficient here? It's used in other tests.